espnet2/asr/encoder/e_branchformer_encoder.py

```diff
@@ -468,11 +522,13 @@ def forward(
 
         intermediate_outs = []
         if len(self.interctc_layer_idx) == 0:
-            if max_layer is not None and 0 <= max_layer < len(self.encoders):
-                for layer_idx, encoder_layer in enumerate(self.encoders):
-                    xs_pad, masks = encoder_layer(xs_pad, masks)
-                    if layer_idx >= max_layer:
-                        break
+            for layer_idx, encoder_layer in enumerate(self.encoders):
+                xs_pad, masks = encoder_layer(xs_pad, masks)
+                if return_all_hs:
+                    if isinstance(xs_pad, tuple):
+                        intermediate_outs.append(xs_pad[0])
+                    else:
+                        intermediate_outs.append(xs_pad)
             else:
                 xs_pad, masks = self.encoders(xs_pad, masks)
```

Comment:
@wanchichen 
I found this for-else makes encoders forward pass doubled while trying XEUS model. Is it intended or bug?

Reply:
Oh this is a bug. Good catch! I will need to check how this impacts performance

Reply:
@wanchichen any update on this? i wanna try XEUS as feature extractor